proxy.h

```diff
@@ -38,6 +50,21 @@ template <class T, class...> struct lazy_eval_traits : std::type_identity<T> {};
 template <class U, class... T>
 using lazy_eval_t = typename lazy_eval_traits<U, T...>::type;
 
+enum class qualifier_type { lv, const_lv, rv, const_rv };
```

Comment:
What about the `volatile` in cv?

Reply:
Similar with the wording of `std::move_only_function`, `volatile` is not supported by design